test/e2e/throughput_test.go

```diff
@@ -0,0 +1,201 @@
+	// Limit its maxScale to 10 containers, wait for the service to scale down and hit it with concurrent requests.
+	logger = logging.GetContextLogger(t.Name())
+	clients = Setup(t)
+
+	helloWorldNames := test.ResourceNames{
+		Config: test.AppendRandomString(configName, logger),
+		Route:  test.AppendRandomString(routeName, logger),
+		Image:  "observed-concurrency",
+	}
+
+	configOptions := test.Options{
+		ContainerConcurrency: 1,
+	}
+
+	fopt := func(config *v1alpha1.Configuration) {
+		config.Spec.RevisionTemplate.Annotations = make(map[string]string)
+		config.Spec.RevisionTemplate.Annotations["autoscaling.knative.dev/maxScale"] = "10"
+	}
+
+	if _, err := test.CreateConfiguration(logger, clients, helloWorldNames, &configOptions, fopt); err != nil {
+		t.Fatalf("Failed to create Configuration: %v", err)
+	}
+
+	if _, err := test.CreateRoute(logger, clients, helloWorldNames); err != nil {
+		t.Fatalf("Failed to create Route: %v", err)
+	}
+
+	test.CleanupOnInterrupt(func() { TearDown(clients, helloWorldNames, logger) }, logger)
+	defer TearDown(clients, helloWorldNames, logger)
+
+	revision, err := test.WaitForConfigLatestRevision(clients, helloWorldNames)
+	if err != nil {
+		t.Fatalf("Configuration %s was not updated with the new revision: %v", helloWorldNames.Config, err)
+	}
+
+	logger.Info("When the Route reports as Ready, everything should be ready.")
+	if err := test.WaitForRouteState(clients.ServingClient, helloWorldNames.Route, test.IsRouteReady, "RouteIsReady"); err != nil {
+		t.Fatalf("The Route %s was not marked as Ready to serve traffic: %v", helloWorldNames.Route, err)
+	}
+
+	helloWorldRoute, err := clients.ServingClient.Routes.Get(helloWorldNames.Route, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get Route %q of helloworld app: %v", helloWorldNames.Route, err)
+	}
+
+	domain := helloWorldRoute.Status.Domain
+
+	deploymentName := revision + "-deployment"
+
+	logger.Info("Waiting for deployment to scale to zero.")
+	if err := pkgTest.WaitForDeploymentState(
+		clients.KubeClient,
+		deploymentName,
+		test.DeploymentScaledToZeroFunc,
+		"DeploymentScaledToZero",
+		test.ServingNamespace,
+		2*time.Minute); err != nil {
+		fmt.Errorf("Failed waiting for deployment to scale to zero: %v", err)
+	}
+
+	// Hit the service with concurrent requests to provoke the start from 0,
+	// the requests should go through activator, we make sure we receive only 200 responses.
+
+	logger.Info("Waiting for endpoint to serve request")
+
+	endpoint, err := spoof.GetServiceEndpoint(clients.KubeClient.Kube)
+	url := fmt.Sprintf("http://%s/?timeout=100", *endpoint)
+
+	client, err := pkgTest.NewSpoofingClient(clients.KubeClient, logger, domain, test.ServingFlags.ResolvableDomain)
+
+	totalRequests := 1000
+	responseChannel := make(chan *spoof.Response, totalRequests)
+	timeout := 30 * time.Second
+	roundTrip := roundTrip(client, url)
+
+	sendRequests(roundTrip, 1000, totalRequests, responseChannel, timeout, logger, t)
```

Comment:
```suggestion
	sendRequests(roundTrip, 1000 /* concurrency */, totalRequests, responseChannel, timeout, logger, t)
```
For some reason it's not a constant like others.